### PR TITLE
feat: fire Rudder Identifier Periodically

### PR DESCRIPTION
**Pull request by anantjain45823**
## Description of the change

> Sending Rudder Identifier at a regular time-period irrespective of the user activity.

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: deviceModeInit.js [hunk 1]
```diff
@@ -157,23 +152,19 @@ var rudderTracking = (function () {
 
   // TODO: add support for product search
 
+  function sendRudderIdentifierEventPeriodically(cart) {
+    const timeInterval = 50 * 60 * 1000; // 50 mins
+    sendIdentifierToRudderWebhook(cart); // sending rudderIdentifier for the first time
+    setInterval(function () {
+      sendIdentifierToRudderWebhook(cart);
+    }, timeInterval);
```
Comment (ItsSudip) [inline deviceModeInit.js:160]: This is not correct. We need to call it periodically. Here we are calling it once.

Comment (anantjain45823) [reply deviceModeInit.js:160]: This `setinterval`() calls it after every `timeInterval`

Comment (ItsSudip) [reply deviceModeInit.js:160]: Got it

File: deviceModeInit.js [hunk 2]
```diff
@@ -86,16 +86,11 @@ var rudderTracking = (function () {
     fetchCart()
       .then((cart) => {
         const needToUpdateCart = checkCartNeedsToBeUpdated(cart);
-        const needToUpdateCookie = checkCookieNeedsToBeUpdated();
```
Comment (ItsSudip) [inline deviceModeInit.js:89]: Why are you removing this? `needToUpdateCart` will be true only once a new cart has been created for the entire session.

Comment (anantjain45823) [reply deviceModeInit.js:89]: Yes, and only inn that  case only we are calling `sendRudderIdentitiferEventPeriodically` which will keep sending rudderIdentifier at a periodically time interval so there is no need of cookies. Hence no need to check those.

Comment (ItsSudip) [reply deviceModeInit.js:89]: But when we refresh the page does it will be false and will then the periodic calls happen?

Comment (anantjain45823) [reply deviceModeInit.js:89]: No , Incorporated the cookie now

**Review by ItsSudip — APPROVED** [review]

File: deviceModeInit.js [hunk 3]
```diff
@@ -159,20 +160,18 @@ var rudderTracking = (function () {
...
-    const cookieOptions = {
-      action: "set",
-      expire_hr: 2,
-      name: "rs_shopify_cart_identified_at",
-      value: `${Date.now()}`
-    }
-    cookie_action(cookieOptions);
+  function sendRudderIdentifierPeriodically() {
+    getAndSendIdentifierToRudderWebhook();
+    const cookieUpdateFixedInterval = 50 * 60 * 1000; // 50 mins
```
Comment (utsabc) [inline deviceModeInit.js:167]: I thought we decided on much lower time of 5 mins or something

File: deviceModeInit.js [hunk 4]
```diff
@@ -187,15 +186,16 @@ var rudderTracking = (function () {
...
+  function getTimeForCookieUpdate() {
+    const cookieUpdateFixedInterval = 50 * 60 * 1000; // 50 mins
     const currentTime = Date.now();
     const prev_rs_shopify_cart_identified_at = cookie_action({
       action: "get",
       name: "rs_shopify_cart_identified_at",
     });
-    return currentTime - prev_rs_shopify_cart_identified_at > oneHourTimeInMilliSeconds;
+    return cookieUpdateFixedInterval + prev_rs_shopify_cart_identified_at  - currentTime;
   }
```
Comment (utsabc) [inline deviceModeInit.js:197]: Please explain what does this function trying to do

**Review by utsabc — CHANGES_REQUESTED** [review]

File: deviceModeInit.js (hunk 3 above)
Comment (anantjain45823) [reply deviceModeInit.js:167]: We don't need pool time any.more that's the  time after which the rudderIdentifier event will fire automatically periodically

File: deviceModeInit.js (hunk 4 above)
Comment (anantjain45823) [reply deviceModeInit.js:197]: This will return the time after which rudderIdentifier will get fired.

**Review by utsabc — DISMISSED** [review]

File: deviceModeInit.js [hunk 5]
```diff
@@ -187,13 +187,14 @@ var rudderTracking = (function () {
...
+    const thresholdTime = 50 * 60 * 1000; // 50 mins
     const currentTime = Date.now();
-    const prev_rs_shopify_cart_identified_at = cookie_action({
+    const last_updated_at = Number(cookie_action({
       action: "get",
       name: "rs_shopify_cart_identified_at",
-    });
-    return cookieUpdateFixedInterval + prev_rs_shopify_cart_identified_at  - currentTime;
+    }));
+    const timeToUpdate = thresholdTime + last_updated_at - currentTime;
```
Comment (utsabc) [inline deviceModeInit.js:196]:
```suggestion
    const timeToUpdate = thresholdTime - (  currentTime - last_updated_at );
```

**Review by utsabc — CHANGES_REQUESTED** [review]

**Review by utsabc — DISMISSED** [review]

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**